src/xdocs/config_coding.xml

```diff
@@ -3623,6 +3623,200 @@ class TestHashCode {
       </subsection>
     </section>
 
+    <section name="MatchXpath">
+      <p>Since Checkstyle 8.39</p>
+      <subsection name="Description" id="MatchXpath_Description">
+        <p>
+          Evaluates Xpath query and report violation on all matching AST nodes. Current check allows
+          user to implement custom checks using Xpath. If Xpath query is not specified explicitly,
+          then check does nothing.
+        </p>
+        <p>
+          Please read more about Xpath syntax at
+          <a href="https://www.w3schools.com/xml/xpath_syntax.asp"> W3Schools Xpath Syntax</a>.
+          Information regarding Xpath functions can be found at
+          <a href="https://developer.mozilla.org/en-US/docs/Web/XPath/Functions">XSLT/XPath
+            Reference</a>. Note, that <b>@text</b> attribute can used only with token types that
+          are listed in
+          <a href="https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/XpathUtil.java#L101">
```

Comment:
I highly recommend not pointing to master in documentation. We can't guarantee this line position will stay the same as new changes are made. Do we even need to point to code in what should be user friendly documentation?

Reply:
Another approach is to list all supported token types inside documentation. @romani what do you think?

Reply:
lets move all this to special doc page in scope of https://github.com/checkstyle/checkstyle/issues/9074
to not block this Check acceptance.

Reply:
meanwhile we can change it to https://github.com/checkstyle/checkstyle/search?q=%22TOKEN_TYPES_WITH_TEXT_ATTRIBUTE+%3D+Arrays.asList%22

Reply:
done